panic on clicking twice on main pane of Status

**Describe the bug**
panic on clicking twice on main pane of Status 

**To Reproduce**
Steps to reproduce the behavior:

1. Click on "Status" pane
2. Click on 'main' pane
3. Click on "main" pane again
4. Crashes with panic

**Expected behavior**
should not panic

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Version info:**
commit=, build date=, build source=Homebrew, version=0.50.0, os=darwin, arch=arm64, git version=2.39.5 (Apple Git-154)
git version 2.39.5 (Apple Git-154)

**Additional context**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x88 pc=0x1051193c8]

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*MainViewController).onClick(0x1400011ca50, {0x1050a1c8c?, 0x140001f7918?})
        github.com/jesseduffield/lazygit/pkg/gui/controllers/main_view_controller.go:98 +0x38
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*MainViewController).GetMouseKeybindings.func1({0x140000728a0?, 0x140000021c0?})
        github.com/jesseduffield/lazygit/pkg/gui/controllers/main_view_controller.go:63 +0xf8
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).resetKeybindings.(*Gui).SetMouseKeybinding.func2({0x105f1e6c8?, 0x4?})
        github.com/jesseduffield/lazygit/pkg/gui/keybindings.go:518 +0xf4
github.com/jesseduffield/gocui.(*Gui).execMouseKeybindings(0x14000320000, 0x140002b8b48, 0x14000045560, {0x0?, 0x14000045488?})
        github.com/jesseduffield/gocui@v0.3.1-0.20250421160159-82c9aaeba2b9/gui.go:1449 +0x234
github.com/jesseduffield/gocui.(*Gui).onKey(0x14000320000, 0x14000045560)
        github.com/jesseduffield/gocui@v0.3.1-0.20250421160159-82c9aaeba2b9/gui.go:1401 +0x500
github.com/jesseduffield/gocui.(*Gui).handleEvent(0x14000130030?, 0x14000045548?)
        github.com/jesseduffield/gocui@v0.3.1-0.20250421160159-82c9aaeba2b9/gui.go:845 +0x40
github.com/jesseduffield/gocui.(*Gui).processEvent(0x14000320000)
        github.com/jesseduffield/gocui@v0.3.1-0.20250421160159-82c9aaeba2b9/gui.go:799 +0x180
github.com/jesseduffield/gocui.(*Gui).MainLoop(0x14000320000)
        github.com/jesseduffield/gocui@v0.3.1-0.20250421160159-82c9aaeba2b9/gui.go:778 +0x108
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0x14000318008, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        github.com/jesseduffield/lazygit/pkg/gui/gui.go:867 +0x434
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1()
        github.com/jesseduffield/lazygit/pkg/gui/gui.go:873 +0x48
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x14000045928?)
        github.com/jesseduffield/lazygit/pkg/utils/utils.go:90 +0x5c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0x14000318008, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        github.com/jesseduffield/lazygit/pkg/gui/gui.go:872 +0xc4
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(...)
        github.com/jesseduffield/lazygit/pkg/app/app.go:270
github.com/jesseduffield/lazygit/pkg/app.Run({0x105613fb8?, 0x140000ca370?}, 0x140001f6bc0, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}})
        github.com/jesseduffield/lazygit/pkg/app/app.go:48 +0xb0
github.com/jesseduffield/lazygit/pkg/app.Start(0x14000045ef8, {0x0, 0x0})
        github.com/jesseduffield/lazygit/pkg/app/entry_point.go:168 +0x9b0
main.main()
        github.com/jesseduffield/lazygit/main.go:23 +0x98
```

--- a/pkg/gui/context.go
+++ b/pkg/gui/context.go
@@ -357,3 +357,19 @@ func (self *ContextMgr) CurrentPopup() []types.Context {
 		return context.GetKind() == types.TEMPORARY_POPUP || context.GetKind() == types.PERSISTENT_POPUP
 	})
 }
+
+func (self *ContextMgr) NextInStack(c types.Context) types.Context {
+	self.RLock()
+	defer self.RUnlock()
+
+	for i := range self.ContextStack {
+		if self.ContextStack[i].GetKey() == c.GetKey() {
+			if i == 0 {
+				return nil
+			}
+			return self.ContextStack[i-1]
+		}
+	}
+
+	panic("context not in stack")
+}

--- a/pkg/gui/controllers/context_lines_controller.go
+++ b/pkg/gui/controllers/context_lines_controller.go
@@ -131,7 +131,9 @@ func (self *ContextLinesController) currentSidePanel() types.Context {
 	currentContext := self.c.Context().CurrentStatic()
 	if currentContext.GetKey() == context.NORMAL_MAIN_CONTEXT_KEY ||
 		currentContext.GetKey() == context.NORMAL_SECONDARY_CONTEXT_KEY {
-		return currentContext.GetParentContext()
+		if sidePanelContext := self.c.Context().NextInStack(currentContext); sidePanelContext != nil {
+			return sidePanelContext
+		}
 	}
 
 	return currentContext

--- a/pkg/gui/controllers/main_view_controller.go
+++ b/pkg/gui/controllers/main_view_controller.go
@@ -56,21 +56,16 @@ func (self *MainViewController) GetKeybindings(opts types.KeybindingsOpts) []*ty
 func (self *MainViewController) GetMouseKeybindings(opts types.KeybindingsOpts) []*gocui.ViewMouseBinding {
 	return []*gocui.ViewMouseBinding{
 		{
-			ViewName: self.context.GetViewName(),
-			Key:      gocui.MouseLeft,
-			Handler: func(opts gocui.ViewMouseBindingOpts) error {
-				if self.isFocused() {
-					return self.onClick(opts)
-				}
-
-				self.context.SetParentContext(self.otherContext.GetParentContext())
-				self.c.Context().Push(self.context, types.OnFocusOpts{
-					ClickedWindowName:  self.context.GetWindowName(),
-					ClickedViewLineIdx: opts.Y,
-				})
-
-				return nil
-			},
+			ViewName:    self.context.GetViewName(),
+			Key:         gocui.MouseLeft,
+			Handler:     self.onClickInAlreadyFocusedView,
+			FocusedView: self.context.GetViewName(),
+		},
+		{
+			ViewName:    self.context.GetViewName(),
+			Key:         gocui.MouseLeft,
+			Handler:     self.onClickInOtherViewOfMainViewPair,
+			FocusedView: self.otherContext.GetViewName(),
 		},
 	}
 }
@@ -93,11 +88,21 @@ func (self *MainViewController) escape() error {
 	return nil
 }
 
-func (self *MainViewController) onClick(opts gocui.ViewMouseBindingOpts) error {
+func (self *MainViewController) onClickInAlreadyFocusedView(opts gocui.ViewMouseBindingOpts) error {
 	parentCtx := self.context.GetParentContext()
 	if parentCtx.GetOnClickFocusedMainView() != nil {
 		return parentCtx.GetOnClickFocusedMainView()(self.context.GetViewName(), opts.Y)
 	}
+	return nil
+}
+
+func (self *MainViewController) onClickInOtherViewOfMainViewPair(opts gocui.ViewMouseBindingOpts) error {
+	self.context.SetParentContext(self.otherContext.GetParentContext())
+	self.c.Context().Push(self.context, types.OnFocusOpts{
+		ClickedWindowName:  self.context.GetWindowName(),
+		ClickedViewLineIdx: opts.Y,
+	})
+
 	return nil
 }
 
@@ -111,8 +116,4 @@ func (self *MainViewController) openSearch() error {
 	}
 
 	return nil
-}
-
-func (self *MainViewController) isFocused() bool {
-	return self.c.Context().Current().GetKey() == self.context.GetKey()
 }

--- a/pkg/gui/controllers/main_view_controller.go
+++ b/pkg/gui/controllers/main_view_controller.go
@@ -76,7 +76,6 @@ func (self *MainViewController) Context() types.Context {
 
 func (self *MainViewController) togglePanel() error {
 	if self.otherContext.GetView().Visible {
-		self.otherContext.SetParentContext(self.context.GetParentContext())
 		self.c.Context().Push(self.otherContext, types.OnFocusOpts{})
 	}
 
@@ -89,15 +88,14 @@ func (self *MainViewController) escape() error {
 }
 
 func (self *MainViewController) onClickInAlreadyFocusedView(opts gocui.ViewMouseBindingOpts) error {
-	parentCtx := self.context.GetParentContext()
-	if parentCtx.GetOnClickFocusedMainView() != nil {
-		return parentCtx.GetOnClickFocusedMainView()(self.context.GetViewName(), opts.Y)
+	sidePanelContext := self.c.Context().NextInStack(self.context)
+	if sidePanelContext != nil && sidePanelContext.GetOnClickFocusedMainView() != nil {
+		return sidePanelContext.GetOnClickFocusedMainView()(self.context.GetViewName(), opts.Y)
 	}
 	return nil
 }
 
 func (self *MainViewController) onClickInOtherViewOfMainViewPair(opts gocui.ViewMouseBindingOpts) error {
-	self.context.SetParentContext(self.otherContext.GetParentContext())
 	self.c.Context().Push(self.context, types.OnFocusOpts{
 		ClickedWindowName:  self.context.GetWindowName(),
 		ClickedViewLineIdx: opts.Y,

--- a/pkg/gui/controllers/rename_similarity_threshold_controller.go
+++ b/pkg/gui/controllers/rename_similarity_threshold_controller.go
@@ -106,7 +106,9 @@ func (self *RenameSimilarityThresholdController) currentSidePanel() types.Contex
 	currentContext := self.c.Context().CurrentStatic()
 	if currentContext.GetKey() == context.NORMAL_MAIN_CONTEXT_KEY ||
 		currentContext.GetKey() == context.NORMAL_SECONDARY_CONTEXT_KEY {
-		return currentContext.GetParentContext()
+		if sidePanelContext := self.c.Context().NextInStack(currentContext); sidePanelContext != nil {
+			return sidePanelContext
+		}
 	}
 
 	return currentContext

--- a/pkg/gui/controllers/switch_to_focused_main_view_controller.go
+++ b/pkg/gui/controllers/switch_to_focused_main_view_controller.go
@@ -60,19 +60,18 @@ func (self *SwitchToFocusedMainViewController) Context() types.Context {
 }
 
 func (self *SwitchToFocusedMainViewController) onClickMain(opts gocui.ViewMouseBindingOpts) error {
-	return self.focusMainView("main")
+	return self.focusMainView(self.c.Contexts().Normal)
 }
 
 func (self *SwitchToFocusedMainViewController) onClickSecondary(opts gocui.ViewMouseBindingOpts) error {
-	return self.focusMainView("secondary")
+	return self.focusMainView(self.c.Contexts().NormalSecondary)
 }
 
 func (self *SwitchToFocusedMainViewController) handleFocusMainView() error {
-	return self.focusMainView("main")
+	return self.focusMainView(self.c.Contexts().Normal)
 }
 
-func (self *SwitchToFocusedMainViewController) focusMainView(mainViewName string) error {
-	mainViewContext := self.c.Helpers().Window.GetContextForWindow(mainViewName)
+func (self *SwitchToFocusedMainViewController) focusMainView(mainViewContext types.Context) error {
 	mainViewContext.SetParentContext(self.context)
 	if context, ok := mainViewContext.(types.ISearchableContext); ok {
 		context.ClearSearchString()

--- a/pkg/gui/controllers/switch_to_focused_main_view_controller.go
+++ b/pkg/gui/controllers/switch_to_focused_main_view_controller.go
@@ -72,7 +72,6 @@ func (self *SwitchToFocusedMainViewController) handleFocusMainView() error {
 }
 
 func (self *SwitchToFocusedMainViewController) focusMainView(mainViewContext types.Context) error {
-	mainViewContext.SetParentContext(self.context)
 	if context, ok := mainViewContext.(types.ISearchableContext); ok {
 		context.ClearSearchString()
 	}

--- a/pkg/gui/types/context.go
+++ b/pkg/gui/types/context.go
@@ -300,6 +300,7 @@ type IContextMgr interface {
 	CurrentStatic() Context
 	CurrentSide() Context
 	CurrentPopup() []Context
+	NextInStack(context Context) Context
 	IsCurrent(c Context) bool
 	IsCurrentOrParent(c Context) bool
 	ForEach(func(Context))

--- a/pkg/gui/view_helpers.go
+++ b/pkg/gui/view_helpers.go
@@ -152,9 +152,9 @@ func (gui *Gui) postRefreshUpdate(c types.Context) {
 			// just don't rerender the view while searching, on the assumption that users will probably
 			// either search or change their data, but not both at the same time.
 			if !currentCtx.GetView().IsSearching() {
-				parentCtx := currentCtx.GetParentContext()
-				if parentCtx.GetKey() == c.GetKey() {
-					parentCtx.HandleRenderToMain()
+				sidePanelContext := gui.State.ContextMgr.NextInStack(currentCtx)
+				if sidePanelContext != nil && sidePanelContext.GetKey() == c.GetKey() {
+					sidePanelContext.HandleRenderToMain()
 				}
 			}
 		}

